Test failbit when converting boundary ids

roystgnr opened the pull request:
Resolves #24950

Pinging @olinwc to tell me if the message is reasonable; this seems to work well on one of his test cases.

roystgnr committed:
Test failbit when converting boundary ids

Resolves #24950

lindsayad reviewed (commented):
wondering if we can unit test this

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1647744/) on 2093329 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24951/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1647762/) on 2093329 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>abeacb</th>
      <th colspan=4>#24951 209332</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.58%</td>
      <td>85.58%</td>
      <td>-0.00%</td>
      <td>33.33%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>89464</td>
      <td>89464</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15073</td>
      <td>15076</td>
      <td>+3</td>
      <td>2</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24951/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24951/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24951/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 33.33% is less than the suggested 90.0%



This comment will be updated on new commits.

roystgnr committed:
Add test for boundary integer conversion failure

Plus a test for conversion success, more as contrast than out of fear
that that's not covered elsewhere.

roystgnr commented:
> wondering if we can unit test this

Yeah, but I got lazy and just put in a regression test instead.

lindsayad commented on test/tests/meshgenerators/polyline_mesh_generator/tests:
```diff
@@ -37,4 +37,21 @@
+    cli_args = '--mesh-only'
+    exodiff = 'polyline_mesh_generator_good_boundary_in.e'
+    recover = false
+    requirement = "The system shall set a boundary name or id if requested"
```
```suggestion
    requirement = "The system shall set a boundary name or id if requested."
```

lindsayad commented on test/tests/meshgenerators/polyline_mesh_generator/tests:
```diff
@@ -37,4 +37,21 @@
+    issues = '#24950'
+    cli_args = '--mesh-only'
+    recover = false
+    requirement = "The system shall detect any unsatisfiable boundary id requests"
```
```suggestion
    requirement = "The system shall detect any unsatisfiable boundary id requests."
```

roystgnr committed:
Apply suggestions from code review

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

roystgnr committed:
Add missing design, issue entries
